clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateUserCommandImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.command;
+
+import io.camunda.zeebe.client.api.JsonMapper;
+import io.camunda.zeebe.client.api.ZeebeFuture;
+import io.camunda.zeebe.client.api.command.CreateUserCommandStep1;
+import io.camunda.zeebe.client.api.command.FinalCommandStep;
+import io.camunda.zeebe.client.api.response.CreateUserResponse;
+import io.camunda.zeebe.client.impl.http.HttpClient;
+import io.camunda.zeebe.client.impl.http.HttpZeebeFuture;
+import io.camunda.zeebe.client.protocol.rest.UserWithPasswordRequest;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.apache.hc.client5.http.config.RequestConfig;
+
+public final class CreateUserCommandImpl implements CreateUserCommandStep1 {
+
+  private final UserWithPasswordRequest request;
+  private final JsonMapper jsonMapper;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
+
+  public CreateUserCommandImpl(final HttpClient httpClient, final JsonMapper jsonMapper) {
+    this.jsonMapper = jsonMapper;
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+    request = new UserWithPasswordRequest();
+  }
+
+  @Override
+  public FinalCommandStep<CreateUserResponse> requestTimeout(final Duration requestTimeout) {
+    httpRequestConfig.setResponseTimeout(requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
+    return this;
+  }
+
+  @Override
+  public ZeebeFuture<CreateUserResponse> send() {
+    ArgumentUtil.ensureNotNull("username", request.getUsername());
+    ArgumentUtil.ensureNotNull("email", request.getEmail());
+    ArgumentUtil.ensureNotNull("name", request.getName());
+    ArgumentUtil.ensureNotNull("password", request.getPassword());
+    final HttpZeebeFuture<CreateUserResponse> result = new HttpZeebeFuture<>();
```

Comment:
❓ in the UserController for the `/users` endpoint we return with `executeServiceMethodWithNoContentResult`, will there be a user response here to map?

Reply:
It had also confused me but I checked the user-tasks endpoint and client and it seems like it's the way to go. And as I understand the fact that response is empty reflected in CreateUserResponse being an empty interface.

Reply:
Correct, it works because it's an empty response. Otherwise there'd have to be a response mapping in the controller